packages/flutter/test/material/menu_anchor_test.dart

```diff
@@ -2672,6 +2672,72 @@ void main() {
       expect(closed, unorderedEquals(<TestMenu>[TestMenu.mainMenu1, TestMenu.subMenu11]));
       expect(opened, isEmpty);
     });
+
+    testWidgets('Using a ListenableMenuController works', (WidgetTester tester) async {
+      final ListenableMenuController listenableController = ListenableMenuController();
+      addTearDown(listenableController.dispose);
+      final FocusNode focusNode = FocusNode();
+      addTearDown(focusNode.dispose);
+
+      await tester.pumpWidget(
+        MaterialApp(
+          theme: ThemeData(useMaterial3: true),
+          home: Material(
+            child: Directionality(
+              textDirection: TextDirection.ltr,
+              child: MenuAnchor(
+                childFocusNode: focusNode,
+                controller: listenableController,
+                menuChildren: <Widget>[
+                  MenuItemButton(
+                    child: Text(TestMenu.mainMenu0.label),
+                  ),
+                  MenuItemButton(
+                    child: Text(TestMenu.mainMenu1.label),
+                  ),
+                  MenuItemButton(
+                    child: Text(TestMenu.mainMenu2.label),
+                  ),
+                ],
+                builder: (BuildContext context, MenuController controller, Widget? child) {
```

Comment:
I deliberately didn't change the menu anchor builder to be covariant on the menu controller.
If we ever decide that the workaround should be removed and integrated directly, then the covariant would be a point of API breakage.